engines/groovie/script.cpp

```diff
@@ -1322,66 +1418,74 @@ void Script::o_copyrecttobg() {	// 0x37
 
 	// Sanity checks to prevent bad pointer access crashes
 	if (left > right) {
-		warning("COPYRECT left:%d > right:%d", left, right);
+		warning("Groovie::Script: COPYRECT left:%d > right:%d", left, right);
 		// swap over left and right parameters
 		uint16 j;
 		j = right;
 		right = left;
 		left = j;
 	}
 	if (top > bottom) {
-		warning("COPYRECT top:%d > bottom:%d", top, bottom);
+		warning("Groovie::Script: COPYRECT top:%d > bottom:%d", top, bottom);
 		// swap over top and bottom parameters
 		uint16 j;
 		j = bottom;
 		bottom = top;
 		top = j;
 	}
 	if (top < baseTop) {
-		warning("COPYRECT top < baseTop... clamping");
+		warning("Groovie::Script: COPYRECT top < baseTop... clamping");
 		top = baseTop;
 	}
 	if (top >= 480) {
-		warning("COPYRECT top >= 480... clamping");
+		warning("Groovie::Script: COPYRECT top >= 480... clamping");
 		top = 480 - 1;
 	}
 	if (bottom >= 480) {
-		warning("COPYRECT bottom >= 480... clamping");
+		warning("Groovie::Script: COPYRECT bottom >= 480... clamping");
 		bottom = 480 - 1;
 	}
 	if (left >= 640) {
-		warning("COPYRECT left >= 640... clamping");
+		warning("Groovie::Script: COPYRECT left >= 640... clamping");
 		left = 640 - 1;
 	}
 	if (right >= 640) {
-		warning("COPYRECT right >= 640... clamping");
+		warning("Groovie::Script: COPYRECT right >= 640... clamping");
 		right = 640 - 1;
 	}
 
 	uint16 i, width = right - left, height = bottom - top;
 	uint32 offset = 0;
 	byte *fg, *bg;
+	uint32 pitch = _vm->_graphicsMan->_foreground.pitch;
 
-	debugC(1, kDebugScript, "COPYRECT((%d,%d)->(%d,%d))", left, top, right, bottom);
+	debugC(1, kDebugScript, "Groovie::Script: COPYRECT((%d,%d)->(%d,%d))", left, top, right, bottom);
+	debugC(2, kDebugVideo, "Groovie::Script: @0x%04X: COPYRECT((%d,%d)->(%d,%d))",_currentInstruction-9, left, top, right, bottom);
 
 	fg = (byte *)_vm->_graphicsMan->_foreground.getBasePtr(left, top - baseTop);
 	bg = (byte *)_vm->_graphicsMan->_background.getBasePtr(left, top - baseTop);
 	for (i = 0; i < height; i++) {
-		memcpy(bg + offset, fg + offset, width);
-		offset += 640;
+		memcpy(bg + offset, fg + offset, width * _vm->_graphicsMan->_foreground.format.bytesPerPixel);
+		offset += pitch;
 	}
-	_vm->_system->copyRectToScreen(_vm->_graphicsMan->_background.getBasePtr(left, top - baseTop), 640, left, top, width, height);
+	// _vm->_graphicsMan->_background.copyFrom(_vm->_graphicsMan->_foreground);
+
+	//{
+	//	Graphics::Surface *_bg = &_vm->_graphicsMan->_background;
+	//	_vm->_system->copyRectToScreen(_bg->getPixels(), _bg->pitch, 0, (_vm->_system->getHeight() - _bg->h) / 2, _bg->w, _bg->h);
+	//}
```

Comment:
Why is this code left here?

Reply:
good question, I think that was from when @Alphard-o was working on it, maybe they can give a reason why we might need to return that, otherwise I'll clean it up